### PR TITLE
Output the channel specification.

### PR DESCRIPTION
**Pull request by mbakke**
Hi!

This replaces the 'describe' step with one that stores the channel specification as an output.

This output can be saved and used as an input for a later run (e.g. to check reproducibility or restore an artifact).

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -63,6 +67,10 @@ runs:
         echo "$HOME/.config/guix/current/bin" >> $GITHUB_PATH
       shell: bash
       name: Set PATH
-    - run: guix describe -f channels-sans-intro
+    # Note: substitute newlines to work-around GitHubs single line limitation.
+    # Taken from <https://github.community/t/set-output-truncates-multiline-strings/16852/10>.
+    - run: |
+        echo "::set-output name=channels::$(guix describe -f channels | python3 -c 'import sys; print(sys.stdin.read().replace("\\","\\\\").replace("\"","\\\"").replace("\n","\\n"))')"
```
Comment (PromyLOPh) [inline action.yml]: If I understand correctly this quoting magic is mainly required for the `echo` above. Can’t we get away with `| tr '\n' ' '` (to make sure `::set-output` is a single line) and use a heredoc above?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -79,3 +79,9 @@ jobs:
              %default-channels)
       - name: Build from external channel
         run: guix build python-jupyterlab
+      # Test that the channels output can be pulled.  This operation should
+      # complete quickly.
+      - name: Build with time machine
+        run: |
+          echo -ne "${{ steps.install-guix.outputs.channels }}" > ${{ runner.temp }}/channels.scm
```
Comment (PromyLOPh) [inline .github/workflows/test.yml]: Maybe like this?

```bash
cat <<EOF > ${{ runner.temp }}/channels.scm
${{ steps.install-guix.outputs.channels }}
EOF
```

**Comment by mbakke** [comment]
Using a heredoc with a single `tr` substitution works great, thanks.  Done in the latest commit.

**Comment by PromyLOPh** [comment]
Looks good, thank you very much!

**Comment by PromyLOPh** [comment]
Hm, actually, the CI failed: https://github.com/PromyLOPh/guix-install-action/runs/3017491893?check_suite_focus=true

**Comment by mbakke** [comment]
Whoops, fixed in #9.  Apologies for the churn!

**Pull request merged**